src/app/routes/getInitialData/index.js

```diff
@@ -18,7 +20,47 @@ const baseUrl = onClient()
   ? getBaseUrl(window.location.origin)
   : process.env.SIMORGH_BASE_URL;
 
-const getUrl = pathname => `${baseUrl}${pathname.replace(ampRegex, '')}.json`;
+export const validateRendererEnvironment = environment => {
+  const validRendererEnvironments = ['test', 'live'];
+
+  if (
+    !(
+      environment &&
+      validRendererEnvironments.includes(environment.toLowerCase())
+    )
+  ) {
+    logger.warn(
+      `Invalid parameter value [${environment}]. Usage: renderer_env=${validRendererEnvironments.join(
+        '|',
+      )}`,
+    );
+  }
+};
+
+const getPathFromUrl = url => {
+  return url.split(/[?]/)[0];
+};
+
+const checkAndAppendParams = pathname => {
+  const url = getPathFromUrl(pathname);
+  const params = qs.parse(pathname.substring(url.length));
+
+  if (params && params.renderer_env) {
+    params.renderer_env = params.renderer_env.toLowerCase();
```

Comment:
We try take an immutable approach when we can - https://wecodetheweb.com/2016/02/12/immutable-javascript-using-es6-and-beyond/ so I wouldn't mutate this object. You also lowercase `renderer_env` here and then lowercase it again in `validateRendererEnvironment`